### PR TITLE
ci: Remove duplicate eslint rule definitions

### PR DESCRIPTION
**Pull request by ryan953**
Favoring the most-strict rules. 

I removed the no-import rules for:
- enzyme
- react-bootstrap
because those npm packages are fully removed from the app, the rule doesnt apply anymore

I also removed `no-danger-with-children` which seemed like a typo of `react/no-danger-with-children`

Next step is to convert the file to eslint.config.js format.
References https://github.com/getsentry/frontend-tsc/issues/82

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -554,12 +557,13 @@ const appRules = {
   'no-restricted-imports': [
     'error',
     {
-      paths: [
+      patterns: [
         {
-          name: 'enzyme',
```
Comment (ryan953) [inline .eslintrc.js:559]: removed enzyme check.

File: .eslintrc.js [hunk 2]
```diff
@@ -554,12 +557,13 @@ const appRules = {
...
-          message:
-            'Please import from `sentry-test/enzyme` instead. See: https://github.com/getsentry/frontend-handbook#undefined-theme-properties-in-tests for more information',
+          group: ['sentry/components/devtoolbar/*'],
```
Comment (ryan953) [inline .eslintrc.js:562]: pulled this up from an override below. The default is to not import from this folder... this this folder has an override so this rule is not applied to files inside the folder.

File: .eslintrc.js [hunk 3]
```diff
@@ -596,11 +599,6 @@ const appRules = {
           message:
             'Optional chaining `?.` and nullish coalescing operators `??` are available and preferred over using `lodash/get`. See https://github.com/getsentry/frontend-handbook#new-syntax for more information',
         },
-        {
-          name: 'react-bootstrap',
```
Comment (ryan953) [inline .eslintrc.js:600]: removed this check.

File: .eslintrc.js [hunk 4]
```diff
@@ -619,6 +617,14 @@ const appRules = {
...
             "Use 'useLocation', 'useParams', 'useNavigate', 'useRoutes' from sentry/utils instead.",
         },
+        {
+          name: 'qs',
+          message: 'Please use query-string instead of qs',
+        },
+        {
+          name: 'moment',
+          message: 'Please import moment-timezone instead of moment',
+        },
```
Comment (ryan953) [inline .eslintrc.js:627]: pulled up from an override below. it should apply to all files

File: .eslintrc.js [hunk 5]
```diff
@@ -827,98 +833,6 @@ module.exports = {
     ...reactRules,
     ...appRules,
     ...strictRules,
-    'react-hooks/rules-of-hooks': 'error',
```
Comment (ryan953) [inline .eslintrc.js:830]: this was already defined above.

File: .eslintrc.js [hunk 6]
```diff
@@ -827,98 +833,6 @@ module.exports = {
...
-    'object-shorthand': 'off',
-    'object-curly-spacing': 'off',
-    'import/no-amd': 'off',
-    'no-danger-with-children': 'off',
-    'no-fallthrough': 'off',
-    'no-obj-calls': 'off',
-    'array-bracket-spacing': 'off',
-    'computed-property-spacing': 'off',
-    'react/no-danger-with-children': 'off',
-    'jest/no-disabled-tests': 'off',
```
Comment (ryan953) [inline .eslintrc.js:921]: these are all set to `error` now.

File: .eslintrc.js [hunk 7]
```diff
@@ -827,98 +833,6 @@ module.exports = {
...
-        ],
-      },
-    ],
-
-    // TODO(@anonrig): Remove this from eslint-sentry-config
-    'space-infix-ops': 'off',
-    'object-shorthand': 'off',
-    'object-curly-spacing': 'off',
-    'import/no-amd': 'off',
-    'no-danger-with-children': 'off',
```
Comment (ryan953) [inline .eslintrc.js:915]: this line `'no-danger-with-children` specifically is removed.

File: .eslintrc.js (hunk 6 above)
Comment (scttcper) [reply .eslintrc.js:921]: i think they were disabled because biome covers these rules. Now we will have two linters doing the same thing

Comment (ryan953) [reply .eslintrc.js:921]: We did yeah. I enabled them to get rid of the duplication in this file and none of them sound like slow checks to add. If someone wanted to argue that eslint was too slow to run before this PR, then it should be taking about the same amount of time as well.

Comment (scttcper) [reply .eslintrc.js:921]: i'm more saying you need to update all the biome ignore with additional eslint ignores. try looking at this with the updated eslint rule https://github.com/getsentry/sentry/blob/master/static/app/views/issueList/customViewsHeader.spec.tsx#L729

Comment (ryan953) [reply .eslintrc.js:921]: oh.. hrm github ci should've caught these then.

File: .eslintrc.js [hunk 8]
```diff
@@ -827,98 +833,6 @@ module.exports = {
...
-    'no-fallthrough': 'off',
-    'no-obj-calls': 'off',
-    'array-bracket-spacing': 'off',
-    'computed-property-spacing': 'off',
```
Comment (scttcper) [inline .eslintrc.js:919]: i think these should stay off, prettier related

File: .eslintrc.js [hunk 9]
```diff
@@ -827,98 +833,6 @@ module.exports = {
...
-            message: 'Please import moment-timezone instead of moment',
-          },
-        ],
-      },
-    ],
-
-    // TODO(@anonrig): Remove this from eslint-sentry-config
-    'space-infix-ops': 'off',
-    'object-shorthand': 'off',
-    'object-curly-spacing': 'off',
```
Comment (scttcper) [inline .eslintrc.js:913]: prettier

File: .eslintrc.js [hunk 10]
```diff
@@ -827,98 +833,6 @@ module.exports = {
...
-          {
-            name: 'moment',
-            message: 'Please import moment-timezone instead of moment',
-          },
-        ],
-      },
-    ],
-
-    // TODO(@anonrig): Remove this from eslint-sentry-config
-    'space-infix-ops': 'off',
```
Comment (scttcper) [inline .eslintrc.js:911]: prettier

File: .eslintrc.js (hunk 8 above)
Comment (ryan953) [reply .eslintrc.js:919]: i'll take a pass through all the prettier rules and add 'off' + comments

File: .eslintrc.js (hunk 6 above)
Comment (ryan953) [reply .eslintrc.js:921]: touched a tsx file to trigger eslint. expecting to see 2 lines inside customViewsHeader.spec.tsx fail this rule.

**Comment by scttcper** [comment]
we should probably just use the config but can also check there for which rules should be disabled https://github.com/prettier/eslint-config-prettier/blob/main/index.js

File: .eslintrc.js (hunk 6 above)
Comment (scttcper) [reply .eslintrc.js:921]: i bet the rules don't match between sentry and getsentry

Comment (ryan953) [reply .eslintrc.js:921]: they basically do! I made an earlier PR to this sentry file that didn't make it into getsentry yet, but they will match 100% with this https://github.com/getsentry/getsentry/pull/15980

and i'll keep it updated with whatever happens in here.

**Review by scttcper — APPROVED** [review]

**Pull request merged**